src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java

```diff
@@ -0,0 +1,316 @@
+            var archiveRecord = archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename);
+            batchItem.setArchiveRecord(archiveRecord);
+        }
+    }
+
+    private boolean shouldAddEntryToManifestFile(BatchItem batchItem) {
+        return equalsAnyStatus(batchItem.getPreviousStatus(), EodHelper.failedArmManifestFileStatus(), EodHelper.failedArmResponseManifestFileStatus());
+    }
+
+    private void writeManifestFile(BatchItems batchItems, File archiveRecordsFile) {
+        archiveRecordFileGenerator.generateArchiveRecords(batchItems.getArchiveRecords(), archiveRecordsFile);
+    }
+
+    protected void copyMetadataToArm(File manifestFile) {
+        try {
+            BinaryData metadataFileBinary = fileOperationService.convertFileToBinaryData(manifestFile.getAbsolutePath());
+            armDataManagementApi.saveBlobDataToArm(manifestFile.getName(), metadataFileBinary);
+        } catch (BlobStorageException e) {
+            if (e.getStatusCode() == BLOB_ALREADY_EXISTS_STATUS_CODE) {
+                log.info("Metadata BLOB already exists", e);
+            } else {
+                log.error("Failed to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+                throw e;
+            }
+        } catch (Exception e) {
+            log.error("Unable to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+            throw e;
+        }
+    }
+
+    private void recoverByUpdatingEodToFailedArmStatus(BatchItem batchItem) {
+        if (batchItem.getArmEod() != null) {
+            batchItem.undoManifestFileChange();
+            if (!batchItem.isRawFilePushNotNeededOrSuccessfulWhenNeeded()) {
+                updateExternalObjectDirectoryStatusToFailed(batchItem.getArmEod(), EodHelper.failedArmRawDataStatus());
+            } else {
+                updateExternalObjectDirectoryStatusToFailed(batchItem.getArmEod(), EodHelper.failedArmManifestFileStatus());
+            }
+        }
+    }
+
+    /**
+     * Contains info related to the processing of a batch item.
+     */
+    @Data
+    static class BatchItem {
+
+        private ExternalObjectDirectoryEntity sourceEod;
+        private ExternalObjectDirectoryEntity armEod;
+        private String previousManifestFile;
+        private ObjectRecordStatusEntity previousStatus;
+        private Boolean rawFilePushSuccessful;
+        private ArchiveRecord archiveRecord;
+
+        public void setArmEod(ExternalObjectDirectoryEntity armEod) {
+            this.armEod = armEod;
+            this.previousManifestFile = armEod.getManifestFile();
+            this.previousStatus = armEod.getStatus();
+        }
+
+        public void undoManifestFileChange() {
+            this.armEod.setManifestFile(this.previousManifestFile);
+        }
+
+        public boolean isRawFilePushNotNeededOrSuccessfulWhenNeeded() {
+            return rawFilePushSuccessful == null || rawFilePushSuccessful;
```

Comment:
Predominately people now use nonNull instead for null checks but for this Boolean you could use Boolean.TRUE.equals(rawFilePushSuccessful)

Reply:
that won't work in my case, because when `rawFilePushSuccessful` I need to get a `true`. Because I need to distinguish when the raw push is not attempted because not needed